tests/components/ambient_network/test_sensor.py

```diff
@@ -0,0 +1,163 @@
+
+    sensor = hass.states.get("sensor.station_a_monthly_rain")
+    assert sensor is not None
+    assert sensor.state == "0.0"
+
+    sensor = hass.states.get("sensor.station_a_irradiance")
+    assert sensor is not None
+    assert sensor.state == "37.64"
+
+    sensor = hass.states.get("sensor.station_a_temperature")
+    assert sensor is not None
+    assert float(sensor.state) == pytest.approx(28.27777)
+
+    sensor = hass.states.get("sensor.station_a_uv_index")
+    assert sensor is not None
+    assert sensor.state == "0"
+
+    sensor = hass.states.get("sensor.station_a_weekly_rain")
+    assert sensor is not None
+    assert sensor.state == "0.0"
+
+    sensor = hass.states.get("sensor.station_a_wind_direction")
+    assert sensor is not None
+    assert sensor.state == "11"
+
+    sensor = hass.states.get("sensor.station_a_wind_gust")
+    assert sensor is not None
+    assert float(sensor.state) == pytest.approx(14.75768)
+
+    sensor = hass.states.get("sensor.station_a_wind_speed")
+    assert sensor is not None
+    assert float(sensor.state) == pytest.approx(14.03347)
+
+
+@freeze_time("2023-11-09")
+@pytest.mark.parametrize("config_entry", ["AA:AA:AA:AA:AA:AA"], indirect=True)
+async def test_sensors_with_stale_data(
+    hass: HomeAssistant, open_api: OpenAPI, config_entry, aioambient
+) -> None:
+    """Test that the sensors are not populated if the data is stale."""
+    await setup_platform(hass, open_api, config_entry)
+
+    sensor = hass.states.get("sensor.station_a_absolute_pressure")
+    assert sensor is None
+
+
+@freeze_time("2023-11-08")
+@pytest.mark.parametrize("config_entry", ["BB:BB:BB:BB:BB:BB"], indirect=True)
+async def test_sensors_with_no_data(
+    hass: HomeAssistant, open_api: OpenAPI, config_entry, aioambient
+) -> None:
+    """Test that the sensors are not populated if the last data is absent."""
+    await setup_platform(hass, open_api, config_entry)
+
+    sensor = hass.states.get("sensor.station_b_absolute_pressure")
+    assert sensor is None
+
+
+@freeze_time("2023-11-08")
+@pytest.mark.parametrize("config_entry", ["CC:CC:CC:CC:CC:CC"], indirect=True)
+async def test_sensors_with_no_update_time(
+    hass: HomeAssistant, open_api: OpenAPI, config_entry, aioambient
+) -> None:
+    """Test that the sensors are not populated if the update time is missing."""
+    await setup_platform(hass, open_api, config_entry)
+
+    sensor = hass.states.get("sensor.station_c_absolute_pressure")
+    assert sensor is None
+
+
+@freeze_time("2023-11-08")
+@pytest.mark.parametrize("config_entry", ["AA:AA:AA:AA:AA:AA"], indirect=True)
+async def test_sensors_disappearing(
+    hass: HomeAssistant, open_api: OpenAPI, config_entry, aioambient, caplog
+) -> None:
+    """Test that we log errors properly."""
+
+    # Normal state, sensor is available.
+    await setup_platform(hass, open_api, config_entry)
+    coordinator = hass.data[ambient_network.DOMAIN][config_entry.entry_id]
```

Comment:
Don't touch internals. Please use the freezer and async_fire_time_changed to change time and trigger a "natural" refresh

Reply:
Fixed.